### ISSUE TITLE
trainGPU bootstraps on opponent-perspective next_state after the Q-seat acts

### ISSUE THREAD

**Issue by cerredz** [cerredz/Pulselib#19]
## Intent
`trainGPU.py` updates the Q-network using the immediate post-step observation as `next_state`. In this environment that observation usually belongs to the next seat to act, not the Q-agent. The trainer is therefore learning from opponent-perspective state vectors instead of the Q-seat's next decision state.

## Problem Description
Current behavior:
- `q_mask` is computed before the step, selecting games where the current cursor seat equals `q_seat`.
- After `env.step(actions)`, `next_state` is the immediate next observation from the environment cursor.
- On ordinary non-terminal actions, that cursor has advanced to another seat, so `next_state` contains another player's hole cards and perspective.

Expected behavior:
- The Q-learning target for the Q-agent should be built from the Q-agent's next decision state, not the opponent's immediate response state.

Conditions:
- This manifests on any Q-agent action that does not leave the Q-agent to act again immediately.
- It is a mainline training bug in a multi-agent turn-based environment.

## Affected Files
- `scripts/Poker/trainGPU.py` - selects Q-agent actions with a pre-step mask but uses the immediate post-step observation as `next_state`.
- `environments/Poker/PokerGPU.py` - returns observations from the current cursor seat, so post-step observations naturally change perspective when turn order advances.

## Code Snippets
```python
# scripts/Poker/trainGPU.py - lines 47-60
actions.fill_(0)
q_mask = (info['seat_idx'] == q_seat)
build_actions(state, actions, info['seat_idx'], rotated_agents, rotated_types, device)
next_state, rewards, dones, truncated, info = env.step(actions)
terminated_before = terminated.clone()
terminated |= dones
active_games = q_mask & ~terminated_before
agents[agent_types.index(PokerAgentType.QLEARNING)].train_step(
    states=state[active_games],
    actions=actions[active_games],
    rewards=rewards[active_games],
    next_states=next_state[active_games],
    dones=dones[active_games]
)
```
The trainer filters by who acted before the step, but it does not roll the environment forward to the Q-agent's next turn before building `next_states`.

```python
# environments/Poker/PokerGPU.py - lines 139-145
self.obs[:, 5:7] = self.hands[self.g, self.idx]
self.obs[:, 7:8] = self.stages.unsqueeze(1)
self.obs[:, 8:9] = ((self.idx - self.button) % self.active_players).unsqueeze(1)
self.obs[:, 9:10] = self.pots.unsqueeze(1)
self.obs[:, 10:11] = (self.highest - self.current_round_bet[self.g, self.idx]).unsqueeze(1)
self.obs[:, 11:12] = self.stacks[self.g, self.idx].unsqueeze(1)
```
The observation is explicitly built from the current cursor seat `self.idx`, so changing `self.idx` changes the owner of the observation.

```python
# environments/Poker/PokerGPU.py - lines 512-514
continue_mask = ~self.is_round_over & has_next_player
if continue_mask.any():
    self.idx[continue_mask] = candidate_seats[continue_mask, next_positions[continue_mask]]
```
Ordinary steps advance the cursor to the next player before returning the next observation.

## Root Cause Analysis
The root cause is a mismatch between single-agent Q-learning assumptions and a turn-based multi-agent environment. `trainGPU.py` records a transition when the Q-seat acts, but it uses the immediate post-step observation as `next_state`. In `PokerGPU`, that observation is defined from `self.idx`, which usually points to the next player after the action. So the trainer mixes a Q-agent action with an opponent-perspective successor state.

## Impact Assessment
- Affected flows: essentially every non-terminal Q-agent update.
- User impact: the Q-network learns TD targets from another player's private cards and seat-relative features.
- Data integrity risk: critical. This breaks the semantic meaning of the state representation used for bootstrapping.
- Failure mode: silent. The tensors are valid and match expected shape, but belong to the wrong player.
- Severity: Critical. This is a direct corruption of the RL transition model.

## Potential Fixes
**Option A - Advance the environment until the Q-seat acts again before recording `next_state`**
Treat opponent turns as part of the environment transition and only record the next observation when control returns to the Q-agent (or the hand ends). This matches turn-based RL semantics best, but requires a more complex rollout loop.

**Option B - Reframe training to a shared-policy seat-agnostic setup**
If the intent is to train on any acting seat, remove the fixed-seat assumption and train on all seats with a consistent state encoding. This is a larger design change.

**Option C - Add a trainer-level regression test for observation ownership**
Assert that the `next_state` used in Q-learning still contains the Q-seat's hole cards. This should accompany any fix.

## Reproduction Steps
1. Instantiate `PokerGPU` with `n_players=3`, `max_players=3`, `n_games=1`.
2. Force seat `0` to be the acting Q-seat and assign distinct hole cards:
   - seat `0`: `[11, 24]`
   - seat `1`: `[35, 48]`
3. Clone the current observation and compute `q_mask = (info['seat_idx'] == 0)`.
4. Call `env.step(torch.tensor([1]))` and clone `next_state`.
5. Inspect the exact tensors the trainer would use:
   - `state[active_games, 5:7]` is `[[11, 24]]`
   - `next_state[active_games, 5:7]` is `[[35, 48]]`
6. `next_state` now contains seat `1`'s cards, not the Q-seat's next observation.

## Additional Context
- This bug is distinct from observation-buffer aliasing. Even with explicit `.clone()` calls, the trainer still uses the wrong owner's next observation.
- The issue is especially severe because private hole cards live inside the state vector, so perspective changes are not benign.
